File: 231215/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.py
count = 0

N = int(input())
arr = list(map(int, input().split()))

for s in range(N):
    for e in range(s + 1, N + 1):
        avg = sum(arr[s:e]) / (e - s)

        if avg in arr[s:e]:
            count += 1

print(count)

File: 231215/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.js
const fs = require('fs');
const inputAll = () => fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const print = (...args) => console.log(...args);

let [N, arr] = inputAll();
N = Number(N);
arr = arr.split(' ').map(elm => Number(elm));

let count = 0;

for(let s = 0; s < N; s++){
    for(let e = s + 1; e <= N; e++){
        let tempSum = 0;

        for(let idx = s; idx < e; idx++){
            tempSum += arr[idx];
        }

        const avg = tempSum / (e - s);

        let includeFlag = false;

        for(let idx = s; idx < e; idx++){
            if(arr[idx] === avg){
                includeFlag = true;
                break;
            }
        }

        if(includeFlag) count++;
    }
}

print(count);